### ISSUE TITLE
His Grace ascension is very bugged

### ISSUE THREAD

**Issue by coiax** [tgstation/tgstation#24809]
tl;dr
1) An ascended His Grace who has bloodthirst HIS_GRACE_FALL_ASLEEP will stop attacking people.
2) An ascended His Grace never sleeps, technically, so lings, zombies and borerhosts cannot climb out no more.

HERE IS LINE https://github.com/tgstation/tgstation/blob/20f06bf820c8538223cde5d81811b24693cab1fa/code/game/objects/items/weapons/his_grace.dm#L101

Okay, so when His Grace has eaten a whole bunch of corpses, it "ascends", which makes it gold, gives it a special description, bonus damage, and most importantly means IT NEVER SLEEPS.

Normally a His Grace timeline looks like this.

- His Grace is woken, starts at 1 bloodthirst
- As time goes on, His Grace gets more thirsty, and the more he's eaten, the faster his thirst gain is
- Feeding bodies to His Grace lowers bloodthirst to len(number_of_bodies)
- Eventually, after the traitor has run around murdering people, they can't lower the thirst fast enough, and it crosses a threshold into FRENZY

- A His Grace in a frenzy will start 1) attack its owner, doing damage, stunning and making it be dropped, and 2) lunging at nearby people doing damage, and if they are unconscious or dead, eat them.
- People eaten while His Grace is in a frenzy will actually lower his bloodthirst, but never below the frenzy threshold

- Eventually, when His Grace passes a high level of bloodthirst, he falls back asleep. The toolbox stops shaking, and it can be picked up (and then possibly unlatched, and the cycle continues)

HOWEVER! When His Grace has consumed a body via someone attacking with it, and the number of bodies is >= 20, then he ASCENDS.

An ascended His Grace has `ascended` set to true, which means he will never sleep. This also has the effect of preventing people alive inside His Grace (generally changelings, zombies and borer hosts) from climbing out. The cycle continues normally until he reaches the threshold where he wants to sleep. But he is ascended, and thus cannot sleep. But since he's constantly returning at this section of code, he ceases attacking people.

**Comment by coiax** [comment]
@Xhuis His Grace is your feature originally, and @Militaires you coded ascension. Please use your CODING SKILLS to determine what is CORRECT BEHAVIOUR.

**Comment by Kromgar** [comment]
I feel like an ascended his grace  shouldn't let people escape from it. THe fact lings can crawl out should awaken it because you have to open the latch to get out

**Comment by Xhuis** [comment]
All relevant parts of his have changed since I made this. I'll look into it tomorrow, though.

**Comment by Militaires** [comment]
e-z fix

@xhuis i got it
